hail/src/main/scala/is/hail/expr/ir/EmitCodeBuilder.scala

```diff
@@ -82,24 +84,48 @@ class EmitCodeBuilder(val emb: EmitMethodBuilder[_], var code: Code[Unit]) exten
   }
 
   def memoize(v: EmitCode, name: String): EmitValue = {
+    require(v.pt.isRealizable)
     val l = emb.newEmitLocal(name, v.pt)
     assign(l, v)
     l
   }
 
   def memoize(v: IEmitCode, name: String): EmitValue = {
+    require(v.pt.isRealizable)
     val l = emb.newEmitLocal(name, v.pt)
     assign(l, v)
     l
   }
 
   def memoizeField[T](ec: EmitCode, name: String): EmitValue = {
+    require(ec.pt.isRealizable)
     val l = emb.newEmitField(name, ec.pt)
     l.store(this, ec)
     l
   }
 
+  private[this] def defineNestedUnusedLabels(x: EmitCode): Unit = {
+    x.pv match {
+      case SStreamCode(_, producer) => StreamProducer.markUnused(producer, emb)
```

Comment:
I kind of don't like this but I'm not sure if I can improve it.  What happens if, instead, consumers who are expecting a stream or a realizable value switch on the type there and explicitly do the markUnused.

 > defineNestedUnusedLabels
 > markUnused

Should `markUnused` be called `defineUnused`?